src/core/Form/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,482 @@
+  labelIsVisible: `${baseClassName}_label--visible`,
+  inputAndPickerWrapper: `${baseClassName}_input-and-picker-wrapper`,
+  inputElementContainer: `${baseClassName}_input-element-container`,
+  inputElement: `${baseClassName}_input`,
+  pickerElementContainer: `${baseClassName}_picker-element-container`,
+  pickerButton: `${baseClassName}_picker-button`,
+  pickerButtonDisabled: `${baseClassName}_picker-button--disabled`,
+  pickerIcon: `${baseClassName}_picker-icon`,
+  styleWrapper: `${baseClassName}_wrapper`,
+  statusTextHasContent: `${baseClassName}_statusText--has-content`,
+};
+
+export interface DatePickerProps {
+  /** Enables date picker for date input.
+   * @default false
+   */
+  datePickerEnabled?: boolean;
+  /**
+   * Custom texts for date picker to use instead of language based texts.
+   * <pre>
+   * DatePickerTextProps {
+   *  openButtonLabel?: string;
+   *  selectedDateLabel?: string;
+   *  closeButtonText?: string;
+   *  selectButtonText?: string;
+   *  yearSelectLabel?: string;
+   *  monthSelectLabel?: string;,
+   *  nextMonthButtonLabel?: string;
+   *  prevMonthButtonLabel?: string;
+   *  monthNames?: [
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *  ];
+   *  weekDays?: [
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *  ];
+   *  weekDayAbbreviations?: [
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *    string,
+   *  ];
+   * }
+   * </pre>
+   */
+  datePickerTexts?: DatePickerTextProps;
+  /** Language for date picker default texts. Language supports Finnish, English, and Swedish.
+   * @default 'fi'
+   */
+  language?: 'fi' | 'en' | 'sv';
+  /** Minimum date user can select from date picker. By default first day of month ten years before current date. */
+  minDate?: Date;
+  /** Maximum date user can select from date picker. By default last day of month ten years after current date. */
+  maxDate?: Date;
+  /** Initial date to focus when date picker is opened. By default the current date. */
+  initialDate?: Date;
+  /** Callback fired to disable dates inside the [minDate, maxDate] range. */
+  shouldDisableDate?: (date: Date) => boolean;
+  /**
+   * Enables custom formatting and parsing for date picker. By default date format is like '1.1.2022'.
+   * If value can't be parsed to date, Invalid Date is returned. Invalid Date is a Date, whose time value is NaN.
+   * <pre>
+   * DateAdapter {
+   *   format: (date: Date) => string;
+   *   parse: (value: string) => Date;
+   * }
+   * </pre>
+   */
+  dateAdapter?: DateAdapter;
+}
+export interface DateInputProps
+  extends StatusTextCommonProps,
+    Omit<HtmlInputProps, 'type' | 'onChange'> {
+  /** TextInput container div class name for custom styling. */
```

Comment:
This still reads "TextInput". Even though it fits in a sense, date input is probably clearer.

Reply:
This still reads "TextInput". Even though it fits in a sense, date input is probably clearer.

Reply:
Fixed